core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1240,7 +1226,8 @@ private void deleteInternal(LockedInodePath inodePath, boolean replayed, long op
               String ufsUri = resolution.getUri().toString();
               UnderFileSystem ufs = resolution.getUfs();
               AlluxioURI parentUri = alluxioUriToDel.getParent();
-              if (!safeRecursiveUFSDeletes.containsKey(parentUri)) {
+              if (alluxioUriToDel.equals(inodePath.getUri())
```

Comment:
That is pretty tricky. If this is the case, then the `ufsSynsChecker` should take care of that when `unchecked=true`.